src/modules/scala.rs

```diff
@@ -0,0 +1,172 @@
+use std::fs;
+
+use crate::configs::scala::ScalaConfig;
+use crate::formatter::StringFormatter;
+
+use super::{Context, Module, RootModuleConfig};
+
+use regex::Regex;
+
+const SCALA_VERSION_PATTERN: &str = "version[\\s](?P<version>[0-9\\.\\-A-z]+)";
+pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
+    let mut module = context.new_module("scala");
+    let config: ScalaConfig = ScalaConfig::try_load(module.config);
+
+    let is_scala_project = context
+        .try_begin_scan()?
+        .set_files(&config.detect_files)
+        .set_extensions(&config.detect_extensions)
+        .set_folders(&config.detect_folders)
+        .is_match();
+
+    if !is_scala_project {
+        return None;
+    }
+
+    let parsed = StringFormatter::new(config.format).and_then(|formatter| {
+        formatter
+            .map_meta(|var, _| match var {
+                "symbol" => Some(config.symbol),
+                _ => None,
+            })
+            .map_style(|variable| match variable {
+                "style" => Some(Ok(config.style)),
+                _ => None,
+            })
+            .map(|variable| match variable {
+                "version" => {
+                    let scala_version = get_scala_version(context)?;
+                    Some(Ok(scala_version))
+                }
+                _ => None,
+            })
+            .parse(None)
+    });
+
+    module.set_segments(match parsed {
+        Ok(segments) => segments,
+        Err(error) => {
+            log::warn!("Error in module `scala`:\n{}", error);
+            return None;
+        }
+    });
+
+    Some(module)
+}
+
+fn get_scala_version(context: &Context) -> Option<String> {
+    let scala_command = String::from("scalac");
+    let output = context.exec_cmd(&scala_command.as_str(), &["-version"])?;
```

Comment:
This can be simplified to:
```suggestion
    let output = context.exec_cmd("scalac", &["-version"])?;
```